numpy/lib/nanfunctions.py

```diff
@@ -606,6 +609,145 @@ def nanprod(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
     return np.prod(a, axis=axis, dtype=dtype, out=out, keepdims=keepdims)
 
 
+def nancumsum(a, axis=None, dtype=None, out=None):
+    """
+    Return the cumulative sum of array elements over a given axis treating Not a
+    Numbers (NaNs) as zero.
+
+    Zeros are returned for slices that are all-NaN or empty.
```

Comment:
Mention that the sum does not change when nans are encountered and that any leading nans are replaced by zeros.